consensus/src/pending_votes.rs

```diff
@@ -60,8 +66,8 @@ pub struct PendingVotes {
         HashMap<HashValue /* LedgerInfo digest */, LedgerInfoWithPartialSignatures>,
     /// Tracks all the signatures of the 2-chain timeout for the given round.
     maybe_partial_2chain_tc: Option<TwoChainTimeoutWithPartialSignatures>,
-    /// Map of Author to vote. This is useful to discard multiple votes.
-    author_to_vote: HashMap<Author, Vote>,
+    /// Map of Author to (vote, li_digest, voting_power). This is useful to discard multiple votes.
+    author_to_vote: HashMap<Author, (Vote, HashValue, u64)>,
```

Comment:
Why do you need to store the voting power here? Can't you pass the validator verfier to `log_voting_power` function instead?

Reply:
this is called from process_certificates, and that is called in many places - even if no votes are being processed. (i.e. many tests)

it seems cleaner to have this only if votes are being processed, and keep the signature of process_certificates without ValidatorVerifier

Reply:
Sorry, didn't make it clear. Yes, it doesn't make sense to pass `ValidatorVerifier` to `process_certificates`. What I meant is pass `ValidatorVerifier` in `log_voting_power_if_proposer` function and get the aggregated voting power of the authors who voted instead of storing it here.

Reply:
log_voting_power_if_proposer is called from process_certificates, because that is the place where we change to a new round (and finish with vote collection).

I don't see a different way to get ValidatorVerifier into log_voting_power_if_proposer, unless I pass it into process_certificate